### PR TITLE
IndexedDB: check the ordering of request error events on txn abort

### PR DESCRIPTION
**Pull request by pwnall**

### CODE REVIEW

**Comment by pwnall** [comment]
If I'm not crazy, the CI for this PR should highlight a bug in Chrome.

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/233544922)

# Firefox (nightly) #
Testing web-platform-tests at revision 617a6821cddbcd0dca7dcdd650f2820859e86306
Using browser at version BuildID 20170517100341; SourceStamp 6e3ca5b38f7173b214b10de49e58cb01890bf39d
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5974/IndexedDB/request-abort-ordering.html">/IndexedDB/request-abort-ordering.html</a></summary>


|                         Subtest                          | Results | Messages |
|----------------------------------------------------------|---------|----------|
|                                                          | OK      |          |
| `IndexedDB: request abort events are delivered in order` | PASS    |          |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/233544923)

# Sauce (safari) #
Testing web-platform-tests at revision 617a6821cddbcd0dca7dcdd650f2820859e86306
Using browser at version 10.0
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5974/IndexedDB/request-abort-ordering.html">/IndexedDB/request-abort-ordering.html</a></summary>


|                         Subtest                          | Results |                                                                        Messages                                                                        |
|----------------------------------------------------------|---------|--------------------------------------------------------------------------------------------------------------------------------------------------------|
|                                                          | OK      |                                                                                                                                                        |
| `IndexedDB: request abort events are delivered in order` | FAIL    | `promise_test: Unhandled rejection with value: object "TypeError: store.getAll is not a function. (In 'store.getAll()', 'store.getAll' is undefined)"` |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/233544929)

# Chrome (unstable) #
Testing web-platform-tests at revision 1aa0e7da550a9fa29eee2e8aec59f29a168d000f

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/233544930)

# Sauce (MicrosoftEdge) #

**Comment by pwnall** [comment]
This test passes in Safari 10.1 for me. Checking to see if Sauce Labs has 10.1 support.

**Comment by pwnall** [comment]
I assume `No tests run.` means that Safari 10.1 is not supported. Reverted PR to use Safari 10. Please use a local install of OSX 10.12.4+ if you wish to confirm that the test passes in Safari 10.1.

File: IndexedDB/request-abort-ordering.html [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    const index = store.index('by_author');
+    const cursorRequest = store.openCursor(IDBKeyRange.lowerBound(0));
+
+    return new Promise((resolve, reject) => {
+      cursorRequest.onerror = event => {
+        event.preventDefault();
+        reject(cursorRequest.error);
+      };
+
+      cursorRequest.onsuccess = () => {
```
Comment (inexorabletash) [inline IndexedDB/request-abort-ordering.html]: `testCase.step_func` in case something in here throws?

**Review by inexorabletash — APPROVED** [review]
lgtm...

File: IndexedDB/request-abort-ordering.html (hunk 1 above)
Comment (pwnall) [inline IndexedDB/request-abort-ordering.html]: Done. Thank you for pointing this out!

**Comment by pwnall** [comment]
@inexorabletash Thank you very much for the quick review!

**Pull request merged**